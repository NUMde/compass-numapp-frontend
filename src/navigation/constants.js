/**
 * enum for all route names
 */
const Routes = {
  LANDING: 'Landing',
  LOGIN: 'Login',
  CHECK_IN: 'CheckIn',
  SURVEY: 'Survey',
  ABOUT: 'About',
  WEBVIEW: 'Webview',
  LEGAL_INFORMATION: 'LegalInformation',
};

/**
 * enum for all stacks
 */
const Stacks = {
  SIGNED_IN: 'SignedIn',
  SIGNED_OUT: 'SignedOut',
};

export { Routes, Stacks };
