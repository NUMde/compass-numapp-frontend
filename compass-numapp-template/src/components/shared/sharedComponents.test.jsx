/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { View, Text } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';
import { Routes } from '~navigation/constants';

import Banner from './banner';
import Spinner from './spinner';
import ScrollIndicatorWrapper from './scrollIndicatorWrapper';

describe('Banner', () => {
  const bannerProps = {
    nav: { navigate: jest.fn(), goBack: jest.fn() },
    title: 'title',
    noMenu: false,
    subTitle: 'subtitle',
    isCheckIn: true,
    noWayBack: true,
    noRefresh: false,
    updateUser: jest.fn(),
  };
  it('should render the banner', () => {
    const tree = render(<Banner {...bannerProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the banner with title & without subtitle', () => {
    const tree = render(<Banner {...bannerProps} subTitle="" />);
    expect(tree).toMatchSnapshot();
  });

  it('refresh button should be present & handler called', () => {
    const { getByTestId } = render(<Banner {...bannerProps} isCheckIn />);
    expect(getByTestId('banner_refresh_btn')).toBeTruthy();
    fireEvent.press(getByTestId('banner_refresh_btn'));
    expect(bannerProps.updateUser).toHaveBeenCalled();
  });

  it('back button should be present & navigation called', () => {
    const { getByTestId } = render(
      <Banner {...bannerProps} noWayBack={false} />,
    );
    expect(getByTestId('banner_back_btn')).toBeTruthy();
    fireEvent.press(getByTestId('banner_back_btn'));
    expect(bannerProps.nav.goBack).toHaveBeenCalled();
  });

  it('menu button should be present & navigation called', () => {
    const { getByTestId } = render(
      <Banner {...bannerProps} noWayBack={false} />,
    );
    expect(getByTestId('banner_menu_btn')).toBeTruthy();
    fireEvent.press(getByTestId('banner_menu_btn'));
    expect(bannerProps.nav.navigate).toHaveBeenCalledWith(Routes.ABOUT);
  });

  it('back button should no be present', () => {
    const { queryByTestId } = render(
      <Banner {...bannerProps} noWayBack={true} noRefresh={true} />,
    );
    expect(queryByTestId('banner_back_btn')).not.toBeTruthy();
  });

  it('menu button should no be present', () => {
    const { queryByTestId } = render(<Banner {...bannerProps} noMenu={true} />);
    expect(queryByTestId('banner_menu_btn')).not.toBeTruthy();
  });
});

describe('ScrollIndicatorWrapper', () => {
  it('scrollIndicatorWrapper should be rendered', () => {
    const demoContent = (
      <View>
        <Text testID="demoContent">Some content here</Text>
      </View>
    );
    const { toJSON, getByTestId } = render(
      <ScrollIndicatorWrapper>{demoContent}</ScrollIndicatorWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('demoContent')).toBeTruthy();
  });
});

describe('Spinner', () => {
  it('should render the spinner', () => {
    const tree = render(<Spinner />);
    expect(tree).toMatchSnapshot();
  });
});
