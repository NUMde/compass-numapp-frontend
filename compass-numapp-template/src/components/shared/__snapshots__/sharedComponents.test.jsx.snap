// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render the banner 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
      "height": 354.06976744186045,
      "paddingTop": 30,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "process": [Function],
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": 354.06976744186045,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 20,
        "width": "90%",
      }
    }
  >
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityHint="Re-fetches the questionnaire and status. The current page will be updated."
        accessibilityLabel="refresh"
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="banner_refresh_btn"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#EB576A",
                  "fontSize": 48.83720930232558,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            󰑐
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 33.75,
            "lineHeight": 43.125,
          }
        }
      >
        title
      </Text>
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans",
            "fontSize": 30,
            "lineHeight": 39.375,
          }
        }
      >
        subtitle
      </Text>
    </View>
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityHint="Opens new page with additional information and links."
        accessibilityLabel="Additional Information"
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="banner_menu_btn"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#EB576A",
                  "fontSize": 48.83720930232558,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            󰍜
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "process": [Function],
        }
      }
      style={
        {
          "alignSelf": "center",
          "bottom": 7.848837209302324,
          "maxHeight": 247.02541914548405,
          "maxWidth": 650,
          "position": "relative",
        }
      }
    />
  </View>
</View>
`;

exports[`Banner should render the banner with title & without subtitle 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
      "height": 354.06976744186045,
      "paddingTop": 30,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "process": [Function],
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": 354.06976744186045,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 20,
        "width": "90%",
      }
    }
  >
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityHint="Re-fetches the questionnaire and status. The current page will be updated."
        accessibilityLabel="refresh"
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="banner_refresh_btn"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#EB576A",
                  "fontSize": 48.83720930232558,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            󰑐
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 33.75,
            "lineHeight": 43.125,
          }
        }
      >
        title
      </Text>
    </View>
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityHint="Opens new page with additional information and links."
        accessibilityLabel="Additional Information"
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="banner_menu_btn"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#EB576A",
                  "fontSize": 48.83720930232558,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            󰍜
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "process": [Function],
        }
      }
      style={
        {
          "alignSelf": "center",
          "bottom": 13.081395348837209,
          "maxHeight": 308.781773931855,
          "maxWidth": 650,
          "position": "relative",
        }
      }
    />
  </View>
</View>
`;

exports[`ScrollIndicatorWrapper scrollIndicatorWrapper should be rendered 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
      }
    }
    enableOnAndroid={true}
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <View>
      <View>
        <Text
          testID="demoContent"
        >
          Some content here
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Spinner should render the spinner 1`] = `
<Modal
  hardwareAccelerated={false}
  testID="Spinner"
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "#1F2532",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        color="#EB576A"
        size="large"
        style={
          {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
